### PR TITLE
Added Hindi sentiment analysis dataset

### PR DESCRIPTION
**Pull request by Akash190104**
Sentiment Analysis Dataset for hindi. Read the sentences to realize it consists of reviews. 

## Checklist for adding MMTEB dataset

Reason for dataset addition:

Covers sentiment Analysis in Hindi which was not covered before.

- [x] I have tested that the dataset runs with the `mteb` package.
- [x] I have run the following models on the task (adding the results to the pr). These can be run using the `mteb run -m {model_name} -t {task_name}` command.
  - [x] `sentence-transformers/paraphrase-multilingual-MiniLM-L12-v2`
  - [x] `intfloat/multilingual-e5-small`
- [x] I have checked that the performance is neither trivial (both models gain close to perfect scores) nor random (both models gain close to random scores).
- [x] I have considered the size of the dataset and reduced it if it is too big (2048 examples is typically large enough for most tasks)
- [x] I have filled out the metadata object in the dataset file (find documentation on it [here](https://github.com/embeddings-benchmark/mteb/blob/main/docs/adding_a_dataset.md#2-creating-the-metadata-object)).
- [x] Run tests locally to make sure nothing is broken using `make test`. 
- [x] Run the formatter to format the code using `make lint`. 
- [x] I have added points for my submission to the [points folder](https://github.com/embeddings-benchmark/mteb/blob/main/docs/mmteb/points.md) using the PR number as the filename (e.g. `438.jsonl`).

### CODE REVIEW

File: mteb/tasks/Classification/hin/SentimentAnalysisHindi.py [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+        date=None,
+        form=["written"],
+        dialect=[],
+        domains=["Reviews"],
+        task_subtypes=["Sentiment/Hate speech"],
+        license=None,
+        socioeconomic_status=None,
+        annotations_creators=None,
+        text_creation=None,
+        bibtex_citation=None,
```
Comment (isaac-chung) [inline mteb/tasks/Classification/hin/SentimentAnalysisHindi.py:30]: Maybe
```
@misc{OdiaGenAI,
  author = {Shantipriya Parida and Sambit Sekhar and Soumendra Kumar Sahoo and Swateek Jena and Abhijeet Parida and Satya Ranjan Dash and Guneet Singh Kohli},
  title = {OdiaGenAI: Generative AI and LLM Initiative for the Odia Language},
  year = {2023},
  publisher = {Hugging Face},
  journal = {Hugging Face repository},
  howpublished = {\url{https://huggingface.co/OdiaGenAI}},
}
```

File: mteb/tasks/Classification/hin/SentimentAnalysisHindi.py [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+        category="s2s",
+        eval_splits=["train"],
+        eval_langs=["hin-Deva"],
+        main_score="f1",
+        date=None,
+        form=["written"],
+        dialect=[],
+        domains=["Reviews"],
+        task_subtypes=["Sentiment/Hate speech"],
+        license=None,
```
Comment (isaac-chung) [inline mteb/tasks/Classification/hin/SentimentAnalysisHindi.py:26]:
```suggestion
        license="CC BY-NC-SA 4.0",
```

File: mteb/tasks/Classification/hin/SentimentAnalysisHindi.py [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+        dataset={
+            "path": "OdiaGenAI/sentiment_analysis_hindi",
+            "revision": "1beac1b941da76a9c51e3e5b39d230fde9a80983",
+        },
+        type="Classification",
+        category="s2s",
+        eval_splits=["train"],
+        eval_langs=["hin-Deva"],
+        main_score="f1",
+        date=None,
```
Comment (isaac-chung) [inline mteb/tasks/Classification/hin/SentimentAnalysisHindi.py:21]: Rough estimate based on versions?

File: mteb/tasks/Classification/hin/SentimentAnalysisHindi.py [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+        form=["written"],
+        dialect=[],
+        domains=["Reviews"],
+        task_subtypes=["Sentiment/Hate speech"],
+        license=None,
+        socioeconomic_status=None,
```
Comment (isaac-chung) [inline mteb/tasks/Classification/hin/SentimentAnalysisHindi.py:27]:
```suggestion
        socioeconomic_status="mixed",
```

File: mteb/tasks/Classification/hin/SentimentAnalysisHindi.py [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+        annotations_creators=None,
+        text_creation=None,
```
Comment (isaac-chung) [inline mteb/tasks/Classification/hin/SentimentAnalysisHindi.py:29]:
```suggestion
        text_creation="found",
```

File: mteb/tasks/Classification/hin/SentimentAnalysisHindi.py [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+        eval_langs=["hin-Deva"],
+        main_score="f1",
+        date=None,
+        form=["written"],
+        dialect=[],
+        domains=["Reviews"],
+        task_subtypes=["Sentiment/Hate speech"],
+        license=None,
+        socioeconomic_status=None,
+        annotations_creators=None,
```
Comment (isaac-chung) [inline mteb/tasks/Classification/hin/SentimentAnalysisHindi.py:28]:
```suggestion
        annotations_creators="derived",
```

**Review by isaac-chung — CHANGES_REQUESTED** [review]
Thanks for your contribution! Just a few comments.
Feel free to add 2 points for the datasets and 2 points to me for the review.

**Comment by Akash190104** [comment]
Added changes. Please take a look @isaac-chung. Thanks!

**Review by isaac-chung — COMMENTED** [review]
Sorry I missed it the first round. Could you please also add the results from running `mteb run -m {model_name} -t {task_name}` with the models listed in the PR checklist? 

Otherwise it looks good. Thanks again!

**Comment by Akash190104** [comment]
I didn't lint properly the first time. Could you please enable auto-merge again.
Sorry for the inconvenience.

**Pull request merged**